### PR TITLE
fix(facade): surface an unbuildable fillet as an error, not a WASM trap

### PR DESCRIPTION
**Pull request by andymai**
## Problem

Filleting the reported shelled box now produces an ordinary operation failure instead of a WebAssembly trap.

Issue #348 reproduces with a 40 by 30 by 20 box, its top face removed, a 2mm inward shell, and a radius 2 fillet on one rim edge. The reporter saw `fillet: null function or function signature mismatch` on occt-wasm 4.3.0 and 5.3.2.

With function names retained, the fault occurs in `Geom2dAdaptor_Curve::EvalD1`, under `BRepFilletAPI_MakeFillet::Build`. The call path passes through `Adaptor3d_CurveOnSurface`, `BRepBlend_SurfRstConstRad`, `BRepBlend_SurfRstLineBuilder`, `ChFi3d_Builder`, and `ChFi3d_FilBuilder`. The 20 frame stack contains no repetition, so this is not a stack overflow. `SAFE_HEAP=2` reports a segmentation fault at the same frame, indicating an access in the null page.

`Geom2dAdaptor_Curve::EvalD0`, `EvalD1`, `EvalD2`, `EvalD3`, and `EvalDN` fall back to `myCurve->EvalDx()` for `OtherCurve` and eval representations. A default-constructed adaptor reaches these calls with a null handle.

The fix belongs in OCCT rather than the facade, because a WebAssembly trap cannot be caught from C++ and there is nothing the facade can wrap it in. Its `Standard_Failure` handler never runs. Native OCCT can convert the equivalent signal through `OSD::SetSignal`, but WebAssembly has no signals and `OcctKernel` calls `OSD::SetSignal(false)`.

## Fix

The `occt` submodule moves from `055a9a8` to fork commit `c7f60ff` on `andymai/OCCT` branch `wasm-patches-v6`. The patch adds a file-local helper in `Geom2dAdaptor_Curve.cxx` that raises `Standard_NullObject` for a null handle, then applies it at all 21 evaluation dereferences.

The guard raises unconditionally instead of using `Standard_NullObject_Raise_if`, because `No_Exception` compiles that macro out of Release builds, which is how the dereference stayed unguarded. The patch is written to be upstreamable: it uses a file-local helper, changes no header, and does not change behavior for a loaded adaptor.

`ChFi3d_Builder` already handles a failure from the blend solver. `Build` now reports `IsDone() == false`, allowing the facade to throw `fillet: operation failed`.

The builder image at `ghcr.io/andymai/occt-wasm-builder:latest` is rebuilt and pushed because CI links its baked OCCT static libraries. The `WASI build + stale-check` job requires `crate/src/occt-wasm.wasm.br` to be re-embedded from its fresh artifact with `scripts/reembed-wasmbr.sh`.

## Validation

The same facade objects change from `RuntimeError: memory access out of bounds` to `std::runtime_error: fillet: operation failed`. The history path reports `filletWithHistory: operation failed`.

Native OCCT 8.0.1 returns `IsDone() == false` for the repro and survives radii from 0.5 to 5.0. The native GTest suite is identical with and without the patch: 7275 tests from 535 suites, 7271 passed, and 4 skipped.

`test/edge-cases.test.ts` adds unbuildable fillet coverage that rejects any `WebAssembly.RuntimeError`. It fails against the unpatched library with `WASM trap "memory access out of bounds"` and passes with the patch. A second test verifies that a subsequent good fillet has volume between 7140 and 7152.

The 40 round plain and history loops also pass, with every good fillet in that volume band. On 5.3.2 the trap ends its current call while the module remains usable afterward, as observed by the reporter and confirmed on both unpatched and patched builds.

---
## Summary by cubic
Fixes unbuildable fillet operations trapping the WASM module when the fillet radius matches the wall thickness. OCCT's blend solver dereferences a null curve handle in `Geom2dAdaptor_Curve`, which becomes a WASM trap no C++ handler can catch; the OCCT fork now raises `Standard_NullObject` there, so the operation reports `IsDone() == false` and the facade throws `fillet: operation failed` like other failures.

**Details**
- Re-embeds `crate/src/occt-wasm.wasm.br` from CI's own artifact, since a local rebuild is not byte-identical.
- Adds tests in `test/edge-cases.test.ts` asserting a catchable failure and a still-usable kernel, with selectors verified to match real geometry so they cannot pass on an invalid shape ID.
- Documents in `CLAUDE.md` why WASM traps escape the facade and OCCT's `*_Raise_if` guards are compiled away in Release builds.

<sup>Written for commit 513e8b51395e0de47c04877cb1ac826623f5c582. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/andymai/occt-wasm/pull/352?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: occt [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 055a9a8a2b3fbb33da2ec9a5445d2b97ffbcd765
+Subproject commit c7f60ffe85f28c67358d1771454044f0521f2fdd
```
[the remark this answers is not in the document]
Comment (andymai) [reply occt:1]: The finding is correct, and the stale blob is expected at this point. I refresh it at the end of review because a local rebuild is not byte-identical to CI's output. The failing WASI build + stale-check job uploads its blob, which `scripts/reembed-wasmbr.sh` commits. I will do this before merge, so the crate ships the fixed OCCT.

File: test/edge-cases.test.ts [hunk 2]
```diff
@@ -600,6 +617,78 @@ describe("type mismatches", () => {
...
+        const edges = kernel.getSubShapes(shape, "edge");
+        let found = 0;
+        for (let i = 0; i < edges.size(); i++) {
+            if (match(kernel.getBoundingBox(edges.get(i), false))) {
+                found = edges.get(i);
+                break;
+            }
+        }
+        edges.delete();
+        return found;
```
[the remark this answers is not in the document]
Comment (andymai) [reply test/edge-cases.test.ts:660]: Accepted and fixed in `af6146e`. A fillet on ID 0 fails with `Invalid shape ID`, which the assertion accepted, so the test could pass without exercising the intended geometry. `edgeWhere` and the top face lookup now assert that they matched before returning. After the change, `test/edge-cases.test.ts` reports 46 passed and 1 skipped.
